exporter/exporterhelper/metricshelper.go

```diff
@@ -31,6 +31,7 @@ type PushMetricsData func(ctx context.Context, td consumerdata.MetricsData) (dro
 type metricsExporter struct {
 	exporterName    string
 	pushMetricsData PushMetricsData
+	stopFunc        Stop
```

Comment:
I am not sure I like the proliferation of stopFunc everywhere. I was hoping this PR will reduce the numbe of places we have stopFunc variables, but it seemed to increase it.

Perhaps one way is to into a bit more refactoring and either eliminating PushXXXData and replacing it with XXXExporter or making PushXXXData a struct which implements XXXExporter interface.

I am not totally sure about which approach would be better, but the current implementation doesn't look very clean to me. It seem to replace somewhat not very nice code with another chunk of not very nice code.

Simply judging from line count change, I'd say this makes things worse by adding more code instead of fixing the bad code.

I'd advise to think a bit more about how to make this nicer. I like the original idea to add Stop() to Exporter interface but I think implementation is sub-optimal.

Reply:
Judging changes based on line count solely is counter-productive and ignores the increased code quality and reliability that interfaces provide.

The comment has two components:
1. Extra code from adding a method to an interface
2. ExporterHelpers may/may not be the cleanest. 

**Extra code from adding a method to an interface**
Adding a method to an interface will always do that with the great benefit of catching issues at compile time. Ie This issue would never have happened https://github.com/open-telemetry/opentelemetry-service/issues/244 and we are very lucky that it didn't happen in production 
And one could argue that this is a workaround for unsafe code 
https://github.com/open-telemetry/opentelemetry-service/blob/master/service/builder/exporters_builder.go#L175
 
Since it is expected to call Stop() during shutdown on all exporters, this belongs in the interface for Exporters.
https://github.com/open-telemetry/opentelemetry-service/blob/master/service/builder/exporters_builder.go#L46-L50
Currently, the Builder holds the stop method for the exporter.

**ExporterHelpers may/may not be the cleanest**
One thing I can do to improve the current change is add a default stop function and provide an option to override it during initialization. That will clean up some things; however, there will still be some Stop functions. 
Re: Push***Data() maybe that isn't the neatest way of doing things; however, this pr doesn't impact that and only highlights what was already an issue. An issue has been created (#261 ) to investigate refactoring and improving ExporterHelper.


**Next Steps**
- I'll update my approach for the stop functions in ExporterHelpers to have a default method and an option to override it. This will places where an empty stop function are created.
- Created Issue #261 to track refactoring ExporterHelpers
- Continue with this approach.

Reply:
You should ignore the counter-productive comments :-)

Anyway, I think the main offenders now are exporter helpers. I'd want to see these refactored as well as see Factory changes that you are planning to do to have a fuller picture of how the entire refactoring will look like in the end.